components/authz-service/server/policy.go

```diff
@@ -762,3 +764,32 @@ func (s *policyServer) logPolicies(policies []*storage.Policy) {
 	}
 	s.log.WithFields(kv).Info("Policy definition")
 }
+
+// requiredFieldsAndProjects verifies that all the required inputs are not empty
+func requiredFieldsAndProjects(obj interface {
+	GetId() string
+	GetName() string
+	GetProjects() []string
+}, resourceName string) error {
+	err := validate.RequiredIDandName(obj, resourceName)
+	if err != nil {
+		return err
+	}
+	return requiredProjects(obj.GetProjects())
+}
+
+// requiredProjects verifies that the projects are not empty and do not include '(unassigned)'
+func requiredProjects(projects []string) error {
+	for _, project := range projects {
+		if validate.EmptyOrWhitespaceOnlyRE.MatchString(project) {
+			e := fmt.Sprintf("projects must contain at least one non-whitespace character")
+			return errors.New(e)
+		}
+		if project == constants.UnassignedProjectID {
+			return errors.Errorf("%q cannot explicitly be set. "+
+				"If you wish to create an object in %q, you should pass no projects on creation.",
```

Comment:
"create an object in (unassigned)" sounds awkward. Perhaps:
"(unassigned) cannot explicitly be set; supply an empty projects array instead."